spec/models/scheduled_event_spec.rb

```diff
@@ -56,4 +57,61 @@
+      expect(array.length).to eq 1
+      expect(array).to eq [schedule_event_with_empty_recurring_rules]
+    end
+
+    it "should get 3 occurences" do
+      array = scheduled_event_recurring_rules.calendar_events(scheduled_event_recurring_rules.start_time, 4)
+      expect(array.length).to eq 1
+      expect(array.collect(&:start_time)).to eq [scheduled_event_recurring_rules.start_time]
+    end
+
+    it "should not reschedule for weekday for any offset" do
+      friday_offset_1 = FactoryGirl.create(:scheduled_event, :offset_1, :start_on_friday)
+      events = friday_offset_1.calendar_events(friday_offset_1.start_time, friday_offset_1.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Friday'
+    end
+
+    it "should reschedule to Tuesday for Sunday scheduled Event with offset 2" do
+      friday_offset_2 = FactoryGirl.create(:scheduled_event, :offset_2, :start_on_friday)
+      events = friday_offset_2.calendar_events(friday_offset_2.start_time, friday_offset_2.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Friday'
+    end
+
+    it "should reschedule to Monday for Sunday scheduled Event with offset 1" do
+      sunday_offset_1 = FactoryGirl.create(:scheduled_event, :offset_1, :start_on_sunday)
+      sunday_offset_1.update_attributes(recurring_rules: "{\"interval\":1,\"until\":null,\"count\":null,\"validations\":{\"day_of_week\":{},\"day_of_month\":[22]},\"rule_type\":\"IceCube::MonthlyRule\",\"week_start\":0}")
+      events = sunday_offset_1.calendar_events(sunday_offset_1.start_time, sunday_offset_1.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Monday'
+    end
+
+    it "should reschedule to Monday for Sunday scheduled Event with offset 1" do
+      sunday_offset_2 = FactoryGirl.create(:scheduled_event, :offset_1, :start_on_sunday)
+      sunday_offset_2.update_attributes(recurring_rules: "{\"interval\":1,\"until\":null,\"count\":null,\"validations\":{\"day_of_week\":{},\"day_of_month\":[22]},\"rule_type\":\"IceCube::MonthlyRule\",\"week_start\":0}")
+      events = sunday_offset_2.calendar_events(sunday_offset_2.start_time, sunday_offset_2.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Monday'
+    end
+
+    it "should reschedule to Thursday for Saturday scheduled Event with offset 4" do
+      saturday_offset_1 = FactoryGirl.create(:scheduled_event, :offset_4, :start_on_saturday)
+      saturday_offset_1.update_attributes(recurring_rules: "{\"interval\":1,\"until\":null,\"count\":null,\"validations\":{\"day_of_week\":{},\"day_of_month\":[22]},\"rule_type\":\"IceCube::MonthlyRule\",\"week_start\":0}")
+      events = saturday_offset_1.calendar_events(saturday_offset_1.start_time, saturday_offset_1.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Thursday'
+    end
+
+    it "should reschedule to Thursday for Saturday scheduled Event with offset 4" do
```

Comment:
this scenario and blow doing same thing. I would change one to :start_on_sunday.